### ISSUE TITLE
quantize.py with tp > 1 gives me multiple npz files e.g llama_tp8_rank2.npz - how do you use all of them in build.py?

### ISSUE THREAD

**Issue by phind-justin** [NVIDIA/TensorRT-LLM#1186]
### System Info

system info does not seem relevant here but can provide more info later - using 8 h100 80gb

0.7.1

So after using examples/quantization/quantize.py on a llama with tp = 8 so this code runs:

```
        export_model_config(
            model,  # The quantized model.
            "llama",  # The type of the model as str, e.g gptj, llama or gptnext.
            map_args_to_torch_dtype[args.dtype],  # The exported weights data type as torch.dtype.
            args.export_path,  # The directory where the exported files will be stored.
            8,  # The number of GPUs used in the inference time for tensor parallelism.
        )
```

I get multiple files in my export path 
```
llama_tp8.json  llama_tp8_rank0.npz  llama_tp8_rank1.npz  llama_tp8_rank2.npz  llama_tp8_rank3.npz  llama_tp8_rank4.npz...
```

but how do i use all 8?? am i supposed to? or does it *just work* with only using one
the next step is to build the model using these quantized_fp8 npz files right?
but how can i pass in multiple to the build function?

```
python ./tensorrt_llm/examples/llama/build.py ... \
        --quantized_fp8_model_path llama_tp1_rank0.npz
```

it seems like the param is a str, not an array

```
    parser.add_argument(
        '--quantized_fp8_model_path',
        type=str,
        default=None,
        help='Path of a quantized model checkpoint in .npz format')
```

### Who can help?

@Tracin the quantization god

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

N/A this is a script question

### Expected behavior

N/A this is a script question

### actual behavior

N/A this is a script question

### additional notes

N/A this is a script question

**Comment by Tracin** [comment]
How about give the folder contains the npz files?

**Comment by phind-justin** [comment]
```
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /app/tensorrt_llm/examples/llama/build.py:902 in <module>                                        │
│                                                                                                  │
│   899 │   │   logger.warning(                                                                    │
│   900 │   │   │   f'Parallelly build TensorRT engines. Please make sure that all of the {args.   │
│   901 │   │   )                                                                                  │
│ ❱ 902 │   │   mp.spawn(build, nprocs=args.world_size, args=(args, ))                             │
│   903 │   else:                                                                                  │
│   904 │   │   args.parallel_build = False                                                        │
│   905 │   │   logger.info('Serially build TensorRT engines.')                                    │
│                                                                                                  │
│ /usr/local/lib/python3.10/dist-packages/torch/multiprocessing/spawn.py:246 in spawn              │
│                                                                                                  │
│   243 │   │   │   " torch.multiprocessing.start_processes(...)" % start_method                   │
│   244 │   │   )                                                                                  │
│   245 │   │   warnings.warn(msg)                                                                 │
│ ❱ 246 │   return start_processes(fn, args, nprocs, join, daemon, start_method="spawn")           │
│   247                                                                                            │
│                                                                                                  │
│ /usr/local/lib/python3.10/dist-packages/torch/multiprocessing/spawn.py:202 in start_processes    │
│                                                                                                  │
│   199 │   │   return context                                                                     │
│   200 │                                                                                          │
│   201 │   # Loop on join until it returns True or raises an exception.                           │
│ ❱ 202 │   while not context.join():                                                              │
│   203 │   │   pass                                                                               │
│   204                                                                                            │
│   205                                                                                            │
│                                                                                                  │
│ /usr/local/lib/python3.10/dist-packages/torch/multiprocessing/spawn.py:163 in join               │
│                                                                                                  │
│   160 │   │   original_trace = self.error_queues[error_index].get()                              │
│   161 │   │   msg = "\n\n-- Process %d terminated with the following error:\n" % error_index     │
│   162 │   │   msg += original_trace                                                              │
│ ❱ 163 │   │   raise ProcessRaisedException(msg, error_index, failed_process.pid)                 │
│   164                                                                                            │
│   165                                                                                            │
│   166 class SpawnContext(ProcessContext):                                                        │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
ProcessRaisedException:

-- Process 3 terminated with the following error:
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/torch/multiprocessing/spawn.py", line 74, in _wrap
    fn(i, *args)
  File "/app/tensorrt_llm/examples/llama/build.py", line 850, in build
    engine = build_rank_engine(builder, builder_config, engine_name,
  File "/app/tensorrt_llm/examples/llama/build.py", line 650, in build_rank_engine
    quant_scales = get_scaling_factors(args.quantized_fp8_model_path,
  File "/app/tensorrt_llm/examples/llama/weight.py", line 75, in get_scaling_factors
    weight_dict = np.load(model_path)
  File "/usr/local/lib/python3.10/dist-packages/numpy/lib/npyio.py", line 405, in load
    fid = stack.enter_context(open(os_fspath(file), "rb"))
IsADirectoryError: [Errno 21] Is a directory:
```

**Comment by phind-justin** [comment]
apparently this is just not available in 0.7.1 but i am testing the main branch scripts

**Comment by hello-11** [comment]
@phind-justin Do you still have the question? If not, we will close it soon.
